management/backup/locations_test.go

```diff
@@ -203,6 +204,166 @@ func TestListLocations(t *testing.T) {
+			},
+		}
+		_, err = client.ChangeLocation(&locations.ChangeLocationParams{
+			Body:    updateBody,
+			Context: pmmapitests.Context,
+		})
+		require.NoError(t, err)
+
+		listResp, err := client.ListLocations(&locations.ListLocationsParams{Context: pmmapitests.Context})
+		require.NoError(t, err)
+
+		checkChange(t, updateBody, listResp.Payload.Locations)
+	})
+
+	t.Run("change config type", func(t *testing.T) {
+		t.Parallel()
+
+		addReqBody := locations.AddLocationBody{
+			Name:        gofakeit.Name(),
+			Description: gofakeit.Question(),
+			PMMServerConfig: &locations.AddLocationParamsBodyPMMServerConfig{
+				Path: "/tmp",
+			},
+		}
+		resp, err := client.AddLocation(&locations.AddLocationParams{
+			Body:    addReqBody,
+			Context: pmmapitests.Context,
+		})
+		require.NoError(t, err)
+		defer deleteLocation(t, client, resp.Payload.LocationID)
+
+		updateBody := locations.ChangeLocationBody{
+			LocationID: resp.Payload.LocationID,
+			Name:       gofakeit.Name(),
+			S3Config: &locations.ChangeLocationParamsBodyS3Config{
+				Endpoint:  "https://example.com",
+				AccessKey: "access_key",
+				SecretKey: "secret_key",
+			},
+		}
+		_, err = client.ChangeLocation(&locations.ChangeLocationParams{
+			Body:    updateBody,
+			Context: pmmapitests.Context,
+		})
+		require.NoError(t, err)
+
+		listResp, err := client.ListLocations(&locations.ListLocationsParams{Context: pmmapitests.Context})
+		require.NoError(t, err)
+
+		checkChange(t, updateBody, listResp.Payload.Locations)
+	})
+
+	t.Run("change to existing name - error", func(t *testing.T) {
+		t.Parallel()
+
+		addReqBody1 := locations.AddLocationBody{
+			Name:        gofakeit.Name(),
+			Description: gofakeit.Question(),
+			PMMServerConfig: &locations.AddLocationParamsBodyPMMServerConfig{
+				Path: "/tmp",
+			},
+		}
+		resp1, err := client.AddLocation(&locations.AddLocationParams{
+			Body:    addReqBody1,
+			Context: pmmapitests.Context,
+		})
+		require.NoError(t, err)
+		defer deleteLocation(t, client, resp1.Payload.LocationID)
+
+		addReqBody2 := locations.AddLocationBody{
+			Name:        gofakeit.Name(),
+			Description: gofakeit.Question(),
+			PMMServerConfig: &locations.AddLocationParamsBodyPMMServerConfig{
+				Path: "/tmp",
+			},
+		}
+		resp2, err := client.AddLocation(&locations.AddLocationParams{
+			Body:    addReqBody2,
+			Context: pmmapitests.Context,
+		})
+		require.NoError(t, err)
+		defer deleteLocation(t, client, resp2.Payload.LocationID)
+
+		updateBody := locations.ChangeLocationBody{
+			LocationID: resp2.Payload.LocationID,
+			Name:       addReqBody1.Name,
+			PMMServerConfig: &locations.ChangeLocationParamsBodyPMMServerConfig{
+				Path: "/tmp",
+			},
+		}
+		_, err = client.ChangeLocation(&locations.ChangeLocationParams{
+			Body:    updateBody,
+			Context: pmmapitests.Context,
+		})
+
+		pmmapitests.AssertAPIErrorf(t, err, 409, codes.AlreadyExists, fmt.Sprintf(`Location with name "%s" already exists.`, updateBody.Name))
```

Comment:
You can remove `fmt.Sprintf` because you already use `pmmapitests.AssertAPIErrorf`